custom/enikshay/reports/filters.py

```diff
@@ -47,6 +47,8 @@ def selected(self):
     @classmethod
     def get_value(cls, request, domain):
         selected = super(EnikshayLocationFilter, cls).get_value(request, domain)
+        if not request.couch_user.has_permission(domain, 'access_all_locations'):
+            selected += request.couch_user.get_location_ids(domain)
```

Comment:
Fixed here: https://github.com/dimagi/commcare-hq/pull/15649/commits/cec4d5d1ab27ca111d9b51fa0316933573799630.

`LocationChoiceProvider.get_choices_for_known_values()` already makes sure that selected locations are accessible. Also, I moved this check to *after* `get_choices_for_known_values()`, because `get_choices_for_known_values()` could return an empty list `selected` contains only non-accessible locations.